src/gilded_rose.js

```diff
@@ -4,6 +4,16 @@ function Item(name, sell_in, quality) {
   this.quality = quality;
 }
 
+Item.prototype = {
+  setStrategy: function(type) {
+    console.log('i am here', type.update_quality.toString())
+    this.type = type
```

Comment:
There is no need for setter or getter functions in JS (in most all cases). Writing `item.strategy = brie` is just fine. No code is needed on item to store new properties.